lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -1785,8 +1785,11 @@ void InterfaceSubContextDelegateImpl::inheritOptionsForBuildingInterface(
   GenericArgs.push_back("-disable-objc-attr-requires-foundation-module");
 
   // If we are supposed to use RequireOSSAModules, do so.
-  genericSubInvocation.getSILOptions().EnableOSSAModules =
-      bool(RequireOSSAModules);
+  if (RequireOSSAModules) {
+    genericSubInvocation.getSILOptions().EnableOSSAModules = true;
```

Comment:
Does this have a test?

Reply:
I added a scan-dependencies test in `[enable-ossa-modules-pass-down.swift](https://github.com/swiftlang/swift/pull/77314#diff-ce6adec4434998d49fa1d6b83e2afbd3ecb263c5d57aabb529ab1463576c7cf2)`.
Does that work ?